common/configuration/db_config_resolver.go

```diff
@@ -0,0 +1,61 @@
+package configuration
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/TBD54566975/ftl/backend/controller/dalerrors"
+	"github.com/TBD54566975/ftl/backend/controller/sql"
+	"github.com/TBD54566975/ftl/internal/slices"
+)
+
+// DBConfigResolver loads values a project's configuration from the given database.
+type DBConfigResolver struct {
+	db sql.DBI
```

Comment:
This should use your new `DAL` interface as well. which will remove the need to do things like `TranslatePGError` hopefully.

Reply:
done!